output/transform/postgres_relations.go

```diff
@@ -253,20 +253,17 @@ func transformPostgresRelations(s snapshot.FullSnapshot, newState state.Persiste
 
 			// Statistic
 			if diffedSchemaStatsExist {
-				indexStats, exists := diffedSchemaStats.IndexStats[index.IndexOid]
-				if exists {
-					statistic := snapshot.IndexStatistic{
-						IndexIdx:    indexIdx,
-						SizeBytes:   indexStats.SizeBytes,
-						IdxScan:     indexStats.IdxScan,
-						IdxTupRead:  indexStats.IdxTupRead,
-						IdxTupFetch: indexStats.IdxTupFetch,
-						IdxBlksRead: indexStats.IdxBlksRead,
-						IdxBlksHit:  indexStats.IdxBlksHit,
-						CachedBytes: index.CachedBytes,
-					}
-					s.IndexStatistics = append(s.IndexStatistics, &statistic)
-				}
+				indexStats := diffedSchemaStats.IndexStats[index.IndexOid]
```

Comment:
We discussed this in https://github.com/pganalyze/collector/pull/633#discussion_r1843234090 too, but I think your plan here is to have `s.IndexInformations` and `s.IndexStatistics` as 1:1.
This sadly still doesn't guarantee it as when `diffedSchemaStatsExist` is somehow false, it'll only create `s.IndexInformations` but not corresponding `s.IndexStatistics`.

As an `else` statement, you may want to add some simple thing like the following for the completeness sake.

```golang
			} else {
				s.IndexStatistics = append(s.IndexStatistics, &snapshot.IndexStatistic{
					IndexIdx:    indexIdx,
				})
			}
```